src/lxc/lxccontainer.c

```diff
@@ -4843,7 +4844,14 @@ static bool do_lxcapi_detach_interface(struct lxc_container *c,
 			}
 		}
 
-		ret = lxc_netdev_move_by_name(ifname, pid_outside, dst_ifname);
+		/* remount /sys so is_wlan() can check if this device is a wlan device. */
+		lxc_attach_remount_sys_proc();
```

Comment:
@brauner is this OK to do or is it a bad idea?

Reply:
You can't do this. Not without a new mount namespace. What you're dong right now is re-mounting proc and sys on the host.

Reply:
@brauner I've updated this to make a call to `unshare(CLONE_NEWNS);` before remounting /sys but after the fork. Thanks

Reply:
You also need to do:
```c
	if (mount(NULL, "/", NULL, MS_REC | MS_PRIVATE, 0) < 0) {
```
to prevent mount propagation from getting in your way.

Reply:
@brauner thanks for the advice, I've added now so that the / path inside the mount namespace is recursively set as a private mount before /sys is remounted, so that any existing mount propagation rules can't take effect inside the new mount namespace.

Reply:
Excellent!